token-swap/program/src/curve/constant_price.rs

```diff
@@ -125,14 +125,14 @@ impl CurveCalculator for ConstantPriceCurve {
             .checked_mul(token_b_price)?
             .checked_add(U256::from(swap_token_a_amount))?;
         let pool_supply = U256::from(pool_supply);
-        match round_direction {
-            RoundDirection::Floor => Some(
+        match liquidity_provider_operation {
+            LiquidityProviderOperation::Withdrawal => Some(
```

Comment:
These need to be flipped as mentioned earlier

Reply:
done